client/rest-high-level/src/main/java/org/elasticsearch/client/ml/EvaluateDataFrameResponse.java

```diff
@@ -0,0 +1,117 @@
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.ml;
+
+import org.elasticsearch.client.ml.dataframe.evaluation.EvaluationMetric;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.xcontent.NamedObjectNotFoundException;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+public class EvaluateDataFrameResponse implements ToXContentObject {
+
+    public static EvaluateDataFrameResponse fromXContent(XContentParser parser) throws IOException {
+        if (parser.currentToken() == null) {
+            parser.nextToken();
+        }
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser::getTokenLocation);
+        ensureExpectedToken(XContentParser.Token.FIELD_NAME, parser.nextToken(), parser::getTokenLocation);
+        String evaluationName = parser.currentName();
+        parser.nextToken();
+        Map<String, EvaluationMetric.Result> metrics = parser.map(LinkedHashMap::new, EvaluateDataFrameResponse::parseMetric);
+        List<EvaluationMetric.Result> knownMetrics =
+            metrics.values().stream()
+                .filter(Objects::nonNull)  // Filter out null values returned by {@link EvaluateDataFrameResponse::parseMetric}.
+                .collect(Collectors.toList());
+        ensureExpectedToken(XContentParser.Token.END_OBJECT, parser.nextToken(), parser::getTokenLocation);
+        return new EvaluateDataFrameResponse(evaluationName, knownMetrics);
+    }
+
+    private static EvaluationMetric.Result parseMetric(XContentParser parser) throws IOException {
+        String metricName = parser.currentName();
+        try {
+            return parser.namedObject(EvaluationMetric.Result.class, metricName, null);
+        } catch (NamedObjectNotFoundException e) {
+            parser.skipChildren();
+            // Metric name not recognized. Return {@code null} value here and filter it out later.
+            return null;
```

Comment:
It seems to me that we should log something when we find one that is not supported by the client, but IS returned via the server. Having a silent failure inhibits the user from being able to see that their client code is out of step with the server code. 

What say you @dimitris-athanasiou

Reply:
Skipping unknown metrics makes the parser lenient and allows integrating old client with new server. It is not very different from other places where we drop unknown fields. It's just it's more visible here as I need to do it explicitly.
Happy to hear more from you on logging best practices. From what I see logger is barely used in client code.

Reply:
Hm, tricky question. Arguably, we couldn't even log in a higher level than `debug` as if the user is performing an action regularly and consciously, we would flood the log. Given the object parser already silently ignores unknown fields, I would be inclined to treat unknown named fields the same way.

Reply:
OK, sounds good to me! 👍